### PR TITLE
Add style from tikui

### PR DESCRIPTION
**Pull request by Gnuk**
Related to #2894

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +42,7 @@
   "devDependencies": {
     "@prettier/plugin-xml": "2.2.0",
     "@rushstack/eslint-patch": "1.1.4",
+    "@tikui/core": "^2.1.0",
```
Comment (pascalgrimaud) [inline package.json:45]: always use fixed version plz

File: package.json [hunk 2]
```diff
@@ -63,6 +65,7 @@
     "prettier-plugin-packagejson": "2.2.18",
     "sass": "1.54.5",
     "sinon": "14.0.0",
+    "tikuidoc-tikui": "^4.0.0",
```
Comment (pascalgrimaud) [inline package.json:68]: same, use fixed version

File: pom.xml [hunk 3]
```diff
@@ -24,8 +24,8 @@
     <java.version>17</java.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <node.version>v16.15.0</node.version>
-    <npm.version>8.5.0</npm.version>
+    <node.version>v16.16.0</node.version>
```
Comment (pascalgrimaud) [inline pom.xml:27]: the last version is 16.17.0, maybe the rebase will solve it

File: .github/actions/setup/action.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ runs:
     - name: 'Setup: Node.js'
       uses: actions/setup-node@v3
       with:
-        node-version: 16.15.0
+        node-version: 16.16.0
```
Comment (pascalgrimaud) [inline .github/actions/setup/action.yml:9]: last version should be 16.17.0

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Comment by hdurix** [comment]
I've got an error when I'm trying to build the app, for instance with `./mvnw`:

```
[INFO] Building on /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style directory.
[INFO] [0] node:internal/modules/cjs/loader:959
[INFO] [0]   throw err;
[INFO] [0]   ^
[INFO] [0] 
[INFO] [0] Error: Cannot find module '/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/node/bin/node_modules/npm/bin/npm-cli.js'
[INFO] [0]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
[INFO] [0]     at Function.Module._load (node:internal/modules/cjs/loader:804:27)
[INFO] [0]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[INFO] [0]     at node:internal/main/run_main_module:17:47 {
[INFO] [0]   code: 'MODULE_NOT_FOUND',
[INFO] [0]   requireStack: []
[INFO] [0] }
[INFO] [0] node:internal/modules/cjs/loader:959
[INFO] [0]   throw err;
[INFO] [0]   ^
[INFO] [0] 
[INFO] [0] Error: Cannot find module '/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/node/bin/node_modules/npm/bin/npx-cli.js'
[INFO] [0]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
[INFO] [0]     at Function.Module._load (node:internal/modules/cjs/loader:804:27)
[INFO] [0]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[INFO] [0]     at node:internal/main/run_main_module:17:47 {
[INFO] [0]   code: 'MODULE_NOT_FOUND',
[INFO] [0]   requireStack: []
[INFO] [0] }
[INFO] [0] npx sass "/home/hippolyte/workspace/jhipster/jhipster-lite/src/main/style":"/home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style" -s compressed --source-map --embed-sources exited with code 1
[INFO] [1] /home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/tikuidoc-tikui/src/documentation/style.scss => /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style/documentation/style.css using SCSS
[INFO] [1] /home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/prismjs/prism.js => /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style/lib/prismjs/prism.js

...

[INFO] [1] /home/hippolyte/workspace/jhipster/jhipster-lite/src/main/style/template/template.pug => /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style/template/template.html using Pug
[INFO] [1] node "/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/@tikui/core/dist/assets-build.js" && node "/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/@tikui/core/dist/pug-build.js" exited with code 0
[INFO] Build failed, first error code found: 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

any idea why?

**Comment by pascalgrimaud** [comment]
@hdurix : need to clean node/ folder as suggested by @Gnuk , but I don't understand why

**Comment by Gnuk** [comment]
> I've got an error when I'm trying to build the app, for instance with `./mvnw`:
> 
> ```
> [INFO] Building on /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style directory.
> [INFO] [0] node:internal/modules/cjs/loader:959
> [INFO] [0]   throw err;
> [INFO] [0]   ^
> [INFO] [0] 
> [INFO] [0] Error: Cannot find module '/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/node/bin/node_modules/npm/bin/npm-cli.js'
> [INFO] [0]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
> [INFO] [0]     at Function.Module._load (node:internal/modules/cjs/loader:804:27)
> [INFO] [0]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
> [INFO] [0]     at node:internal/main/run_main_module:17:47 {
> [INFO] [0]   code: 'MODULE_NOT_FOUND',
> [INFO] [0]   requireStack: []
> [INFO] [0] }
> [INFO] [0] node:internal/modules/cjs/loader:959
> [INFO] [0]   throw err;
> [INFO] [0]   ^
> [INFO] [0] 
> [INFO] [0] Error: Cannot find module '/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/node/bin/node_modules/npm/bin/npx-cli.js'
> [INFO] [0]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
> [INFO] [0]     at Function.Module._load (node:internal/modules/cjs/loader:804:27)
> [INFO] [0]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
> [INFO] [0]     at node:internal/main/run_main_module:17:47 {
> [INFO] [0]   code: 'MODULE_NOT_FOUND',
> [INFO] [0]   requireStack: []
> [INFO] [0] }
> [INFO] [0] npx sass "/home/hippolyte/workspace/jhipster/jhipster-lite/src/main/style":"/home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style" -s compressed --source-map --embed-sources exited with code 1
> [INFO] [1] /home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/tikuidoc-tikui/src/documentation/style.scss => /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style/documentation/style.css using SCSS
> [INFO] [1] /home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/prismjs/prism.js => /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style/lib/prismjs/prism.js
> 
> ...
> 
> [INFO] [1] /home/hippolyte/workspace/jhipster/jhipster-lite/src/main/style/template/template.pug => /home/hippolyte/workspace/jhipster/jhipster-lite/target/classes/public/style/template/template.html using Pug
> [INFO] [1] node "/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/@tikui/core/dist/assets-build.js" && node "/home/hippolyte/workspace/jhipster/jhipster-lite/node_modules/@tikui/core/dist/pug-build.js" exited with code 0
> [INFO] Build failed, first error code found: 1
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> ```
> 
> any idea why?

Yes, `node` directory seems not be removed with `clean`, maybe remove it manually then launch `./mvnw`

**Comment by hdurix** [comment]
better now, thanks. This may be the case for every contributors once we merge this one.

**Comment by Gnuk** [comment]
> better now, thanks. This may be the case for every contributors once we merge this one.

Yes, may we need to link this PR to the contributors during this phase.

**Comment by pascalgrimaud** [comment]
Adding a bounty on this nice PR.
Plz claim it @Gnuk

**Comment by Gnuk** [comment]
Claimed https://opencollective.com/generator-jhipster/expenses/92758

**Comment by pascalgrimaud** [comment]
@Gnuk : approved

**Pull request merged**